litestar/utils/predicates.py

```diff
@@ -39,6 +39,11 @@
 from litestar.utils.helpers import unwrap_partial
 from litestar.utils.typing import get_origin_or_inner_type
 
+if sys.version_info >= (3, 10):
+    from inspect import iscoroutinefunction
+else:
```

Comment:
I think that the coverage is now lower, we need to add
```suggestion
else:  # pragma: no cover
```

here